server/manager/src/main/java/org/apache/accumulo/manager/compaction/coordinator/QueueMetrics.java

```diff
@@ -139,7 +140,16 @@ public void update() {
 
     SetView<CompactorGroupId> metricsWithoutQueues =
         Sets.difference(queuesWithMetrics, definedQueues);
-    metricsWithoutQueues.forEach(q -> {
+    Set<CompactorGroupId> toRemove = new HashSet<>(metricsWithoutQueues.size());
```

Comment:
If you used [copyInto](https://guava.dev/releases/23.0/api/docs/com/google/common/collect/Sets.SetView.html#copyInto-S-) would that still require the cloning functionality for `CompactorGroupId`?